service/controller/resource/app/desired.go

```diff
@@ -110,8 +110,10 @@ func (r *Resource) newApp(appOperatorVersion string, cr apiv1alpha3.Cluster, app
 				Name: key.KubeConfigSecretName(&cr),
 			},
 			Secret: applicationv1alpha1.AppSpecKubeConfigSecret{
-				Name:      key.KubeConfigSecretName(&cr),
-				Namespace: key.ClusterID(&cr),
+				Name: key.KubeConfigSecretName(&cr),
+				// The kubeconfig secret is created in the same namespace as
+				// the cluster CR.
+				Namespace: cr.GetNamespace(),
```

Comment:
We want to create the cluster CRs in the organization namespace and not default as we do now.